eixampledb-api/src/main/java/com/eixampledb/api/MainController.java

```diff
@@ -27,54 +25,36 @@
     public ResponseEntity<String> get(@PathVariable("key") String key) {
         GetResponse getResponse = eixampledb.get(new GetRequest(key));
         if (getResponse.isSuccess()) {
-            return ResponseEntity.ok(getResponse.getEntry().get().getValue());
+            return ResponseEntity.ok(String.valueOf(getResponse.getEntry().get().getValue()));
         } else {
             return ResponseEntity.notFound().build();
         }
     }
 
     @RequestMapping(path = "/{key}", method = RequestMethod.POST)
-    public ResponseEntity set(@PathVariable("key") String key, @RequestBody String value, @RequestHeader Map<String,String> header) {
```

Comment:
It is better to use @RequestHeader on individual headers. Use a default value to prevent the messy if-else you've written.